x-pack/plugins/security_solution/public/management/components/endpoint_responder/components/action_log_button.tsx

```diff
@@ -20,6 +20,22 @@ export const ActionLogButton = memo<EndpointResponderExtensionComponentProps>((p
     });
   }, []);
 
+  const endpointInfo = useMemo(
+    () => ({
+      hostname: props.meta.endpoint
+        ? props.meta.endpoint.host.hostname
+        : props.meta.sentinel_one
+        ? props.meta.sentinel_one.host.name
+        : '',
+      agentId: props.meta.endpoint
+        ? props.meta.endpoint.agent.id
+        : props.meta.sentinel_one
+        ? props.meta.sentinel_one.agent.id
+        : '',
```

Comment:
we should really simplify this. There is no need for every component to do this type of calculation if they need to get the agent id and the host name (or other common pieces of information that any EDR system should be able to provide)

My suggestion is to change the command `meta` for all commands to have the following new attributes:

```
agentId: string;
agentType: ResponseActionAgentType; // you already added this one
hostName: string;
```

and then you can just retrieve them with no conditional checks.